build.zig

```diff
@@ -0,0 +1,175 @@
+// THIS MATERIAL IS PROVIDED AS IS, WITH ABSOLUTELY NO WARRANTY EXPRESSED
+// OR IMPLIED.  ANY USE IS AT YOUR OWN RISK.
+//
+// Permission is hereby granted to use or copy this program
+// for any purpose, provided the above notices are retained on all copies.
+// Permission to modify the code and to distribute modified code is granted,
+// provided the above notices are retained, and a notice that the code was
+// modified is included with the above copyright notice.
+
+const std = @import("std");
+const print = @import("std").debug.print;
+
+pub fn build(b: *std.build.Builder) void {
+    const optimize = b.standardOptimizeOption(.{});
+    const target = b.standardTargetOptions(.{});
+    const t = target.toTarget();
+    var default_enable_threads = true;
+    if (t.isWasm()) { // matches both emscripten and wasi
+        default_enable_threads = false;
+    }
+    const enable_threads = b.option(bool, "enable_threads", "Support threads") orelse default_enable_threads;
+    const enable_parallel_mark = b.option(bool, "enable_parallel_mark", "Parallelize marking and free list construction") orelse true;
+    const enable_thread_local_alloc = b.option(bool, "enable_thread_local_alloc", "Turn on thread-local allocation optimization") orelse true;
+    const enable_gcj_support = b.option(bool, "enable_gcj_support", "Support for gcj") orelse true;
+    const enable_java_finalization = b.option(bool, "enable_java_finalization", "Support for java finalization") orelse true;
+    const enable_atomic_uncollectable = b.option(bool, "enable_atomic_uncollectable", "Support for atomic uncollectible allocation") orelse true;
+    const enable_redirect_malloc = b.option(bool, "enable_redirect_malloc", "Redirect malloc and friend to GC routines") orelse false;
+    const enable_disclaim = b.option(bool, "enable_disclaim", "Support alternative finalization interface") orelse true;
+    const enable_large_config = b.option(bool, "enable_large_config", "Optimize for large heap or root set") orelse true;
+    const enable_mmap = b.option(bool, "enable_mmap", "Use mmap instead of sbrk to expand the heap") orelse true;
+    const enable_munmap = b.option(bool, "enable_munmap", "Return page to the OS if empty for N collections") orelse true;
+
+    const lib = b.addStaticLibrary(.{
+        .name = "gc",
+        .target = target,
+        .optimize = optimize,
+    });
+
+    var flags = std.ArrayList([]const u8).init(b.allocator);
+    defer flags.deinit();
+
+    if (t.abi.isMusl()) {
+        std.log.debug("Musl detected", .{});
+        flags.appendSlice(&.{"-DNO_GETCONTEXT"}) catch unreachable;
+    }
+
+    flags.appendSlice(&.{
+        "-fno-sanitize=undefined",
+        // Always enabled
+        "-DALL_INTERIOR_POINTERS",
```

Comment:
Sort of fixed. I switched away from manipulating raw flags to instead use a function, `lib.defineCMacro()` which I think looks cleaner (and is most certainly the intended to be used for this). That doesn't take `-D` at all..

Reply:
I switched back from using defineCMacro because it is easier to reuse the flags for multiple compilation targets if I don't add them directly to the lib but keep them as a separate list. I needed this to apply the same set of flags for tests as well, thus your original point stands. I have added the space!